backend-models/service-tests/showDuggaservice-test.txt

```diff
@@ -0,0 +1,393 @@
+        $answer = “im updated”
+    }
+
+    SQL-query: "UPDATE userAnswer SET submitted=NOW(), useranswer=$answer, timesSubmitted=timesSubmitted+1 WHERE hash=$hash AND password=$hashpwd
+
+***
+Gather service output
+***
+    Save the values of the echoed array as a JSON (at the end of the file. echo json_encode($array);)
+    This is the expected output for the micro service.
+
+***
+* Remove inserted row from DB, using hash
+***
+
+	SQL-query: DELETE FROM userAnswer WHERE hash = pre-values[2];
+
+—------------------------------------------------------------------------------------------------------------------
+—------------------------------------------------------------------------------------------------------------------
+TEST #6 - 293 - 307
+submit dugga
+
+***
+Pre-req: !isset($grade) || !$grade > 1
+***
+
+***
+* send parameters input to showDuggaservice.php
+***
+    send
+    {
+        $opt = “SAVDU”
+        $cid = 2
+        $coursevers = 52432
+        $duggaid = 1
+        $moment = 2002
+        $variant = 3
+        $hash = “ghj1ghj2”
+        $haspwd = “asddasdd”
+        $answer = NULL
+    }
+
+    SQL-query INSERT INTO userAnswer (cid,quiz,moment,vers,variant,hash,password,timesSubmitted,timesAccessed,useranswer,submitted) VALUES(2, 1, 2002, 52432 , 3, “ghj1ghj2” , “asddasdd”,1 ,1 , NULL ,now());")
+
+***
+get newly created data
+***
+
+	SQL-query: SELECT hash FROM useranswer WHERE hash = $hash
+
+***
+Gather service output
+***
+    Save the values of the echoed array as a JSON (at the end of the file. echo json_encode($array);)
+    This is the expected output for the micro service.
+
+***
+* Remove inserted row from DB, using hash
+***
+
+	SQL-query: DELETE FROM userAnswer WHERE hash = $hash;
+
+—------------------------------------------------------------------------------------------------------------------
+—------------------------------------------------------------------------------------------------------------------
+TEST #7 (rows 319-324)
+super-view data from useranswer on hash
+
+***
+Pre-req: 
+    isSuperUser = TRUE
+    $hash != "UNK"
+***
+
+***
+* send parameters input to showDuggaservice.php
+***
+    send
+    {
+        $hash = hash
+    }
+
+***
+get all data
+***
+    SELECT vid,variant.variantanswer AS variantanswer,useranswer,param,cid,vers,quiz FROM userAnswer LEFT JOIN variant ON userAnswer.variant=variant.vid WHERE hash=hash;
+
+***
+Gather service output
+***
+    Save the values of the echoed array as a JSON (at the end of the file. echo json_encode($array);)
+    This is the expected output for the micro service.
+—------------------------------------------------------------------------------------------------------------------
+—------------------------------------------------------------------------------------------------------------------
+TEST #8 (rows 327-334)
+super-view data from useranswer on moment
+
+***
+Pre-req: 
+    isSuperUser = TRUE
+    $hash != "UNK"
+    $rows = NULL
+***
```

Comment:
true, but which circumstance garrisons that this will be the case ? '
$rows must have the value NULL, therefor we need to know which value $hash has to have to produce no row data from the database.

Reply:
this is where the query: SELECT * FROM userAnswer WHERE hash = “xxxxxxxx” to make sure the requested rows == NULL